internal/exec/resolvable/resolvable.go

```diff
@@ -47,7 +47,7 @@ type Field struct {
 	HasContext        bool
 	HasError          bool
 	ArgsPacker        *packer.StructPacker
-	DirectivesPackers map[string]*packer.StructPacker
+	DirectiveVisitors []directives.ResolverInterceptor
```

Comment:
Yep its something that needs to change. In changes I'm making on top of this to add validators, I've had to replace this so that both kinds of directive handling can be retrieved with the right subset for each place they need to run